en/developer_guide/block_parameter.rst

```diff
@@ -0,0 +1,126 @@
+Block parameter
+===============
+
+Context
+-------
+
+To be properly displayed on the website, some blocks will require environment parameters.
+Those parameters could be of multiple types, for instance :
+
+ - url parameters
+ - server parameters
+ - header parameters
+ - ...
+
+An easy way to give those parameters to the block would have been to give all the request
+parameters when the block is called.
+
+This easy solution is not optimal for a main reason: when loading blocks via `ESI`_, each
+block will be cached for each url variant.
+
+If the block 1 does not require any parameters, in the following exemple :
+
+ * ``/block/1`` will be in one cache entry.
+ * ``/block/1?foo=bar`` will be in another cache entry.
+
+To improve your cache hit ratio, it will be likely to reduce the number of cache
+entry for the same page.
+
+Open Orchestra solution
+-----------------------
+
+Open Orchestra gives you the opportunity to pass only the parameters you want
+to request a block, this way limits the cache entries for a single block.
+
+This solution is decomposed in two part :
+- in the Backoffice, where you add the block
+- in the Frontoffice, where you will require some parameters linked to the block.
+
+Backoffice work
+~~~~~~~~~~~~~~~
+
+In the Backoffice, we provide a way to store the parameter linked to the block in the block reference
+in the areas.
+
+You need to declare a service with the tag ``open_orchestra_backoffice.block_parameter.strategy``
+
+.. code-block:: yaml
+
+    tags:
+        - { name: open_orchestra_backoffice.block_parameter.strategy }
+
+The class should also implement the interface ``OpenOrchestra\Backoffice\BlockParameter\BlockParameterInterface``.
+
+In this interface, the method ``getBlockParameter`` will return an array of all the parameters needed by the block
+to work properly.
+
+Frontoffice work
+~~~~~~~~~~~~~~~~
+
+In the Frontoffice, the request to display a block (in the main request) is created while you don't
+have any idea of the block type. You will have to get the block request arguments based only on the
+data stored in the ``blockParameter`` entry from the block reference.
+
+To analyse the parameters and complete the request, you need to declare a service with the tag ``open_orchestra_front.sub_query.strategy``
+
+.. code-block:: yaml
+
+    tags:
+        - { name: open_orchestra_front.sub_query.strategy }
+
+The class should also implement the interface ``OpenOrchestra\FrontBundle\SubQuery\SubQueryGeneratorInterface``
+
+Example
+-------
+
+Subquery strategies available
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
```

Comment:
I think a little intro should be great here. Something like "Open Orchestra comes with some strategies to provides specific parameters to your blocks"